networking/p2p/src/main/java/tech/pegasys/teku/networking/p2p/connection/ConnectionManager.java

```diff
@@ -103,9 +109,16 @@ public ConnectionManager(
   private void connectToKnownPeers() {
     final int maxAttempts = targetPeerCountRange.getPeersToAdd(network.getPeerCount());
     LOG.trace("Connecting to up to {} known peers", maxAttempts);
+    if (maxAttempts == 0) {
+      return;
+    }
+    final PeerScorer peerScorer = peerScorerFactory.create();
     discoveryService
         .streamKnownPeers()
         .filter(this::isPeerValid)
+        .sorted(
+            Comparator.comparing((Function<DiscoveryPeer, Integer>) peerScorer::scoreCandidatePeer)
+                .reversed())
```

Comment:
May be it makes sense to add some randomness here to make connection manager more reliable? I.e. take 50% of peers by score and 50% of random peers.
It occurred several times from my former experience that strictly selecting peers by their score may lead to connectivity problems due to different reasons

Reply:
Seems like a good idea, logged https://github.com/PegaSysEng/teku/issues/2332 to follow up and do that.